src/store/reducers/board-reducer.js

```diff
@@ -17,95 +22,135 @@ export const CLEAR_ALL_FILTERS = "boards/CLEAR_ALL_FILTERS";
 const initialState = {
   boards: [],
   board: null,
-  isLoading: false,
-  error: null,
+  loading: {},
+  errors: {},
   filterBy: getDefaultFilter(),
 };
 
+const handlers = {
+  ...createAsyncHandlers(ADD_LIST, ADD_LIST.KEY),
+  [ADD_LIST.SUCCESS]: (state, action) => ({
+    ...state,
+    loading: { ...state.loading, [ADD_LIST.KEY]: false },
+    board: {
+      ...state.board,
+      lists: [...state.board.lists, action.payload],
+    },
+  }),
+  ...createAsyncHandlers(MOVE_ALL_CARDS, MOVE_ALL_CARDS.KEY),
+  [MOVE_ALL_CARDS.SUCCESS]: (state, action) => ({
+    ...state,
+    loading: { ...state.loading, [MOVE_ALL_CARDS.KEY]: false },
+    board: {
+      ...state.board,
+      lists: action.payload,
+    },
+  }),
+  ...createAsyncHandlers(ARCHIVE_LIST, ARCHIVE_LIST.KEY),
+  [ARCHIVE_LIST.SUCCESS]: (state, action) => ({
+    ...state,
+    loading: { ...state.loading, [ARCHIVE_LIST.KEY]: false },
+    board: {
+      ...state.board,
+      lists: state.board.lists.filter(list => list.id !== action.payload.id),
+    },
+  }),
+  ...createAsyncHandlers(UNARCHIVE_LIST, UNARCHIVE_LIST.KEY),
+  [SET_BOARDS]: (state, action) => ({
+    ...state,
+    boards: action.payload,
+  }),
+  [SET_BOARD]: (state, action) => ({
+    ...state,
+    board: action.payload,
+  }),
+  [DELETE_BOARD]: (state, action) => {
+    const boards = state.boards.filter(board => board._id !== action.payload);
+    return { ...state, boards };
+  },
+  [ADD_BOARD]: (state, action) => ({
+    ...state,
+    boards: [...state.boards, action.payload],
+  }),
+  [UPDATE_BOARD]: (state, action) => ({
+    ...state,
+    board: action.payload,
+  }),
+  [MOVE_LIST]: (state, action) => ({
+    ...state,
+    board: {
+      ...state.board,
+      lists: action.payload,
+    },
+  }),
+  [ADD_CARD]: (state, action) => ({
+    ...state,
+    board: {
+      ...state.board,
+      lists: state.board.lists.map(list =>
+        list.id === action.payload.listId
+          ? { ...list, cards: [...list.cards, action.payload.card] }
+          : list
+      ),
+    },
+  }),
+  [EDIT_CARD]: (state, action) => ({
+    ...state,
+    board: {
+      ...state.board,
+      lists: state.board.lists.map(list =>
+        list.id === action.payload.listId
+          ? {
+              ...list,
+              cards: list.cards.map(card =>
+                card.id === action.payload.card.id ? action.payload.card : card
+              ),
+            }
+          : list
+      ),
+    },
+  }),
+  [DELETE_CARD]: (state, action) => ({
+    ...state,
+    board: {
+      ...state.board,
+      lists: state.board.lists.map(list =>
+        list.id === action.payload.listId
+          ? {
+              ...list,
+              cards: list.cards.filter(
+                card => card.id !== action.payload.cardId
+              ),
+            }
+          : list
+      ),
+    },
+  }),
```

Comment:
The reducers regarding labels should appear somewhere here ( ADD_LABEL , EDIT_LABEL , DELETE_LABEL , UPDATE_CARD_LABELS ) .